examples/platforms/posix/radio.c

```diff
@@ -308,16 +309,24 @@ void otPlatRadioSetPromiscuous(bool aEnable)
 void posixRadioInit(void)
 {
     struct sockaddr_in sockaddr;
+    char *offset;
     memset(&sockaddr, 0, sizeof(sockaddr));
     sockaddr.sin_family = AF_INET;
 
+    offset = getenv("PORT_OFFSET");
+
+    if (offset)
+    {
+        sPortOffset = strtol(offset, NULL, 0) * WELLKNOWN_NODE_ID;
```

Comment:
0 will be returned for "0", so this won't work.
since EINVAL is not portable, then there is not much we can do here.

Since we want to allow the offset to be anywhere in UDP port domain, saving output to uint16_t does the job.